### ISSUE TITLE
Adding local tracks to playlists

### ISSUE THREAD

**Issue by richjhart** [spotify/web-api#1421]
I'm aware this has been brought up before. But it's been over 2 years since I last found any discussion of it, and I think it's a really important feature. I was incredibly surprised to discover it's not supported already.

I can't see why it would be technically difficult to be able to add local tracks to a playlist (in my case, I only want to add tracks that already exist in another playlist - so the API could easily check that the URI is valid... i.e. it already exists in the cloud in another playlist).

Basically not having this means I'll be pretty much forced to abandon my project, and may even be forced to reconsider using Spotify.

**Comment by DataGhost** [comment]
Same issue here, I don't see any technical reason why it wouldn't be possible except that maybe a corner case needs to be added to check for "local" AND "track" rather than just "track" in the URIs. My project has also become partially useless now and I only found out after writing all the code, because this weird limitation is not obvious, and "hidden" in one line at the end of the docs on local files, and the docs about playlist management suggest no limitations.

I would very much like to be able to add local tracks to my playlists. Even the option to append a whole playlist to a playlist would be a huge help already.

**Comment by singofwalls** [comment]
Even being able to just duplicate playlists (with local files) would allow us to pare them down to the tracks we want and then add non-local songs. No idea if the ability to duplicate playlists with local files would be much easier to implement, though.
